### ISSUE TITLE
Node stuck on UJ status for 3 days

### ISSUE THREAD

**Issue by Soooooda** [scylladb/scylladb#15193]
This is Scylla's bug tracker, to be used for reporting bugs only.
If you have a question about Scylla, and not a bug, please ask it in
our mailing-list at scylladb-dev@googlegroups.com or in our slack channel.

- ☑️ I have read the disclaimer above, and I am reporting a suspected malfunction in Scylla.

*Installation details*
Scylla version (or git commit hash): 5.2.4
Cluster size: 5 nodes (1 node is UJ)
OS (RHEL/CentOS/Ubuntu/AWS AMI): Ubuntu

*Hardware details (for performance issues)*         
AWS EC2 i4i.xlarge "ami-0ce0eca2ef60d90b8" provided by Scylla

report UUID: 27525d60-80cf-4ec1-86eb-68ed4e902ad7

**Comment by Soooooda** [comment]
Found some error messages of the NJ nodes: 
```
[shard 1] sstable - failed reading index for xxx/me-65-big-Data.db: std::bad_alloc (std::bad_alloc)
```
But there's no core dump.

**Comment by bhalevy** [comment]
@Soooooda I see only the log of `ip-10-11-40-112` in `scylla-logs.txt.gz`.
We'll definitely need the log of the node that's stuck joining - 10.11.33.102
and it'd be best to get logs of all nodes to figure out if there's a problem with schema agreement and why.

**Comment by Soooooda** [comment]
@bhalevy  I should have uploaded all the logs using this UUID. I was testing another UUID earlier and there was only log of ip-10-11-40-112 in scylla-logs.txt.gz. 
<img width="1135" alt="image" src="https://github.com/scylladb/scylladb/assets/19390013/48c6029c-eb60-44e7-a856-62fcb03d4f30">

**Comment by Soooooda** [comment]
@bhalevy Can you try this UUID `18a4b599-894d-45bd-9589-56283e66bcce` ?

**Comment by bhalevy** [comment]
got it.

@tgrabiec node-5 (ip-10-11-33-102) seems to be pulling schema endlessly.
Can you please advise?

**Comment by tgrabiec** [comment]
All nodes are at the same version (b6b6193e-a95b-35a3-a0da-825241737d26), so the pulls are probably due to fetches of unknown table schema version, maybe due to https://github.com/scylladb/scylladb/issues/4485. This shouldn't cause bootstrap to hang though, it's a separate (performance) problem. The workaround is to do a rolling restart of the cluster.

**Comment by tgrabiec** [comment]
There is some flakiness of alive status:

```
Aug 27 21:52:09 ip-10-11-33-102 scylla[5531]:  [shard 0] raft_group_registry - marking Raft server 5da00645-b895-4e04-8d9d-3d0bf5fdae09 as dead for raft groups
Aug 27 21:52:09 ip-10-11-33-102 scylla[5531]:  [shard 0] raft_group_registry - marking Raft server 5da00645-b895-4e04-8d9d-3d0bf5fdae09 as alive for raft groups
Aug 27 21:52:11 ip-10-11-33-102 scylla[5531]:  [shard 0] raft_group_registry - marking Raft server 5da00645-b895-4e04-8d9d-3d0bf5fdae09 as dead for raft groups
Aug 27 21:52:12 ip-10-11-33-102 scylla[5531]:  [shard 0] raft_group_registry - marking Raft server 5da00645-b895-4e04-8d9d-3d0bf5fdae09 as alive for raft groups
Aug 27 21:52:14 ip-10-11-33-102 scylla[5531]:  [shard 0] raft_group_registry - marking Raft server 5da00645-b895-4e04-8d9d-3d0bf5fdae09 as dead for raft groups
Aug 27 21:52:16 ip-10-11-33-102 scylla[5531]:  [shard 0] raft_group_registry - marking Raft server 5da00645-b895-4e04-8d9d-3d0bf5fdae09 as alive for raft groups
Aug 27 21:52:17 ip-10-11-33-102 scylla[5531]:  [shard 0] raft_group_registry - marking Raft server 5da00645-b895-4e04-8d9d-3d0bf5fdae09 as dead for raft groups
Aug 27 21:52:18 ip-10-11-33-102 scylla[5531]:  [shard 0] raft_group_registry - marking Raft server 5da00645-b895-4e04-8d9d-3d0bf5fdae09 as alive for raft groups
```

May point to unresponsiveness of nodes due to overload.


There's lots of warnings like this in the bootstrapping node:
```
Aug 27 07:46:21 ip-10-11-33-102 scylla[5531]:  [shard 1] lsa-timing - reclaim took 67999 us, trying to release 18.969 MiB non-preemptibly, reserve: {goal: 1, max: 30}, at 0x537c98e 0x537cf40 0x537d228 0x1c3747d 0x1c3569c 0x1c4aea2 0x4fb05c5 0x4fb0bdf 0x4fb0ff8 0x4fb260d 0x4fafc6e 0x4f9472d 0x4f832ee 0x4f83278 0x4f9295f 0x191f400 0x3e10cff 0x191f7fc 0x4f88cc0 0x4fa50d6 0x4fa4aab 0x4fa40ba 0x1fe1dfb 0x1fdc0b2 0x1fd4333 0x1fd201c 0x1fd197b 0x2173160 0x216fb69 0x217199c 0x1dddd62 0x1ddec23 0x1dead44 0x1de9932 0x1eb9444 0x1ba2e6a 0x1bc5202 0x1b7713b 0x1b7752f 0x1dddd62 0x1ddec23 0x1dead44 0x1de9932 0x35369ce 0x3537863 0x18dd345 0x18fd98d 0x18fbf6d 0x178b3a6 0x18236af 0x1843499 0x17ebbe4 0x17eaa88 0x2c4b592 0x2d1b1fe 0x2d19928 0x2d0aba9 0x2d1de11 0x14f31e2 0x14f1d3a 0x14f13a9 0x14ef3b3 0x14ee751 0x5363a75
                                                 --------
                                                 seastar::internal::do_until_state<seastar::rpc::server::connection::process()::$_22::operator()()::{lambda()#1}::operator()() const::{lambda()#1}, seastar::rpc::server::connection::process()::$_22::operator()()::{lambda()#1}::operator()() const::{lambda()#2}>
                                                 --------
                                                 seastar::continuation<seastar::internal::promise_base_with_type<void>, seastar::rpc::server::connection::process()::$_23, seastar::future<void>::then_wrapped_nrvo<seastar::future<void>, seastar::rpc::server::connection::process()::$_23>(seastar::rpc::server::connection::process()::$_23&&)::{lambda(seastar::internal::promise_base_with_type<void>&&, seastar::rpc::server::connection::process()::$_23&, seastar::future_state<seastar::internal::monostate>&&)#1}, void>
                                                 --------
                                                 seastar::continuation<seastar::internal::promise_base_with_type<void>, seastar::future<void>::finally_body<seastar::rpc::server::connection::process()::$_24, false>, seastar::future<void>::then_wrapped_nrvo<seastar::future<void>, seastar::future<void>::finally_body<seastar::rpc::server::connection::process()::$_24, false> >(seastar::future<void>::finally_body<seastar::rpc::server::connection::process()::$_24, false>&&)::{lambda(seastar::internal::promise_base_with_type<void>&&, seastar::future<void>::finally_body<seastar::rpc::server::connection::process()::$_24, false>&, seastar::future_state<seastar::internal::monostate>&&)#1}, void>
```

**Comment by tgrabiec** [comment]
@asias @denesb I recall we saw some reader deadlock issues related to repair, which could explain the stuck streaming. Do you know if anything is unsolved in 5.2.4?

**Comment by mykaul** [comment]
> @asias @denesb I recall we saw some reader deadlock issues related to repair, which could explain the stuck streaming. Do you know if anything is unsolved in 5.2.4?

Is it https://github.com/scylladb/scylladb/commit/aa5e904c40365109228344d1e171e350c801e64c ? It missed 5.2.4, landed in 5.2.6.

**Comment by tgrabiec** [comment]
@Soooooda Please retest with 5.2.6

**Comment by denesb** [comment]
> > @asias @denesb I recall we saw some reader deadlock issues related to repair, which could explain the stuck streaming. Do you know if anything is unsolved in 5.2.4?
> 
> Is it [aa5e904](https://github.com/scylladb/scylladb/commit/aa5e904c40365109228344d1e171e350c801e64c) ? It missed 5.2.4, landed in 5.2.6.

I think this is a good candidate.

**Comment by Soooooda** [comment]
@denesb Resolved after I upgraded to 5.2.6.
